src/test/java/api/requests/RequestExpiredHoldsResourceTest.java

```diff
@@ -0,0 +1,215 @@
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+public class RequestExpiredHoldsResourceTest extends APITests {
+
+  @Test
+  public void regularRequestForEmptyData()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    final UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+    Response response = ResourceClient.forRequestReport(client).getById(pickupServicePointId);
+
+    assertThat(response.getStatusCode(), is(HTTP_OK));
+
+    JsonObject responseJson = response.getJson();
+    assertThat(responseJson.getInteger("totalRecords"), is(0));
+  }
+
+  @Test
+  public void regularRequestWhenItemIsCheckedOut()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    final InventoryItemResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+
+    loansFixture.checkOutByBarcode(smallAngryPlanet, usersFixture.james());
+    requestsFixture.place(new RequestBuilder()
+      .open()
+      .hold()
+      .withPickupServicePointId(pickupServicePointId)
+      .forItem(smallAngryPlanet)
+      .by(usersFixture.rebecca()));
+
+    Response response = ResourceClient.forRequestReport(client).getById(pickupServicePointId);
+    assertThat(response.getStatusCode(), is(HTTP_OK));
+
+    JsonObject responseJson = response.getJson();
+    assertThat(responseJson.getInteger("totalRecords"), is(0));
+  }
+
+  @Test
+  public void multipleRequestsAwaitingPickupOnItems()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    final InventoryItemResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final InventoryItemResource temeraire = itemsFixture.basedUponTemeraire();
+    final UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+
+    loansFixture.checkOutByBarcode(temeraire, usersFixture.charlotte());
+    requestsClient.create(new RequestBuilder()
+      .open()
+      .hold()
+      .withPickupServicePointId(pickupServicePointId)
+      .forItem(temeraire)
+      .by(usersFixture.steve()));
+    loansFixture.checkInByBarcode(temeraire);
+
+    loansFixture.checkOutByBarcode(smallAngryPlanet, usersFixture.james());
+    requestsClient.create(new RequestBuilder()
+      .open()
+      .hold()
+      .withPickupServicePointId(pickupServicePointId)
+      .forItem(smallAngryPlanet)
+      .by(usersFixture.rebecca()));
+    loansFixture.checkInByBarcode(smallAngryPlanet);
+
+    Response response = ResourceClient.forRequestReport(client).getById(pickupServicePointId);
+    assertThat(response.getStatusCode(), is(HTTP_OK));
+
+    JsonObject responseJson = response.getJson();
+    assertThat(responseJson.getInteger("totalRecords"), is(0));
+  }
+
+  @Test
+  public void multipleClosedPickupExpiredRequest()
```

Comment:
The story refers to the idea that the report includes only requests that were cancelled or expired whilst awaiting pickup.

`Select the request for each item that expired or was cancelled on the hold shelf most recently (by ranking the closed requests by a date used only for this purpose)`

How does this test demonstrate that behaviour, given that we are relying on the storage module to identify this change and detecting it by the presence of the new field?

Reply:
@marcjohnson-kint It is difficult for me to reproduce the real case when the condition is met:
- items that were expired or cancelled while sitting on the hold shelf.

Since in the real case, it is necessary to manage the due date.
I tried to force the case by changing the status of the request in other tests.
But I have to be sure that no items will be included in the report when:
item.status = AwaitingPickup and request.status = "Open - Awaiting pickup"
And I assume that additional tests are not redundant.

Reply:
@OleksiiMaksymov 

I'm not suggesting that the other tests are redundant. I was asking how this test demonstrates the desired behaviour. To me, it demonstrates behaviour that we do not want, by including all `Closed - Pickup expired` requests in the report. 

It is not possible to demonstrate this by state transition alone in mod-circulation.

The solution @kostyantyn-kh proposed relies on using the `awaitingPickupRequestClosedDate ` (introduced in @opochepam [change to mod-circulation-storage](https://github.com/folio-org/mod-circulation-storage/pull/164) for a dual purpose, both to identify the transition from `Open - Awaiting pickup` to `Closed - Cancelled` or `Closed - Pickup expired` and to rank them to find the latest.

In order to demonstrate the report behaviour in mod-circulation, I think the tests have to mimic the behaviour of mod-circulation-storage that we are coupling to.

@kostyantyn-kh what do you think? Can you help advise how we are intending to demonstrate the behaviour given the coupling between these modules.

Reply:
@marcjohnson-kint I understood the problem and will soon make changes to the implementation

Reply:
@marcjohnson-kint  My understanding is we need to rely on results from _mod-circulation-storage_
and check behavior in _mod-circulation_. Create requests with expected statuses and _awaitingPickupRequestClosedDate_ and check expected result in report